pkg/bbgo/source.go

```diff
@@ -0,0 +1,50 @@
+package bbgo
+
+import (
+	"strings"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+	log "github.com/sirupsen/logrus"
+)
+
+type SourceFunc func(*types.KLine) fixedpoint.Value
+
+var Four fixedpoint.Value = fixedpoint.NewFromInt(4)
+var Three fixedpoint.Value = fixedpoint.NewFromInt(3)
+var Two fixedpoint.Value = fixedpoint.NewFromInt(2)
+
+type SourceSelector struct {
+	Source    string `json:"source,omitempty"`
+	getSource SourceFunc
+}
+
+func (s *SourceSelector) Init() {
```

Comment:
You can implement a UnmarshalJSON method, and after you unmarshal the json, you assign the getSource function, this way the Init() method call can be removed.

Reply:
done